src/wallet/rpcwallet.cpp

```diff
@@ -2434,13 +2435,16 @@ static UniValue getbalances(const JSONRPCRequest& request)
 
     UniValue obj(UniValue::VOBJ);
 
+    const bool avoid_reuse = wallet.IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE);
     const auto bal = wallet.GetBalance();
+    const auto bal_used = avoid_reuse ? wallet.GetBalance(0, false) : bal;
```

Comment:
I felt like this logic could have used more commenting, and the variable was a bit confusingly named (and also could potentially be scoped more tightly).

I think the following is equivalent, but perhaps a little clearer:

```
→ git d
diff --git a/src/wallet/rpcwallet.cpp b/src/wallet/rpcwallet.cpp
index ce1c94905..f53469ce6 100644
--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -2435,16 +2435,19 @@ static UniValue getbalances(const JSONRPCRequest& request)
 
     UniValue obj(UniValue::VOBJ);
 
-    const bool avoid_reuse = wallet.IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE);
     const auto bal = wallet.GetBalance();
-    const auto bal_used = avoid_reuse ? wallet.GetBalance(0, false) : bal;
     UniValue balances{UniValue::VOBJ};
     {
         UniValue balances_mine{UniValue::VOBJ};
         balances_mine.pushKV("trusted", ValueFromAmount(bal.m_mine_trusted));
         balances_mine.pushKV("untrusted_pending", ValueFromAmount(bal.m_mine_untrusted_pending));
         balances_mine.pushKV("immature", ValueFromAmount(bal.m_mine_immature));
-        if (avoid_reuse) balances_mine.pushKV("used", ValueFromAmount(bal_used.m_mine_trusted + bal_used.m_mine_untrusted_pending - bal.m_mine_trusted - bal.m_mine_untrusted_pending));
+        if (wallet.IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE)) {
+            // If the AVOID_REUSE flag is set, bal has been set to just the un-reused address balance. Get
+            // the total balance, and then subtract bal to get the reused address balance.
+            const auto full_bal = wallet.GetBalance(0, false);
+            balances_mine.pushKV("used", ValueFromAmount(full_bal.m_mine_trusted + full_bal.m_mine_untrusted_pending - bal.m_mine_trusted - bal.m_mine_untrusted_pending));
+        }
         balances.pushKV("mine", balances_mine);
     }
     if (wallet.HaveWatchOnly()) {
```

Feel free to use if you think that's better.

Reply:
Yep, that looks cleaner. Thanks!